src/client/app/components/maps/MapViewComponent.tsx

```diff
@@ -2,436 +2,72 @@
-				<FormattedMessage id={buttonMessageId} />
-			</Button>;
-		} else {
-			toggleButton = <div />;
-		}
-
-		if (hasToken()) {
-			return (
-				<div>
-					{formattedCircleSize}
-					{toggleButton}
-				</div>
-			);
-		} else {
-			return (
-				<div>
-					{this.props.map.circleSize}
-					{toggleButton}
-				</div>
-			);
-		}
-	}
-
-	private formatStatus(): string {
-		if (this.props.isSubmitting) {
-			return '(' + this.props.intl.formatMessage({ id: 'submitting' }) + ')';
-		}
-		if (this.props.isEdited) {
-			return this.props.intl.formatMessage({ id: 'edited' });
-		}
-		return '';
-	}
-
-	private toggleDelete() {
-		const consent = window.confirm(`${this.props.intl.formatMessage({ id: 'map.confirm.remove' })} "${this.props.map.name}"?`);
-		if (consent) { this.props.removeMap(this.props.id); }
-	}
-
-	private formatDeleteButton() {
-		const editButtonStyle: React.CSSProperties = {
-			display: 'inline', // or 'none'
-			paddingLeft: '5px'
-		};
-		return <Button style={editButtonStyle} color='primary' onClick={this.toggleDelete}>
-			<FormattedMessage id={'delete.map'} />
-		</Button>;
-	}
-
-	private styleEnabled(): React.CSSProperties {
-		return { color: 'green' };
-	}
-
-	private styleDisabled(): React.CSSProperties {
-		return { color: 'red' };
-	}
-
-	private styleToggleBtn(): React.CSSProperties {
-		return { float: 'right' };
-	}
-
-	private toggleMapDisplayable() {
-		const editedMap = {
-			...this.props.map,
-			displayable: !this.props.map.displayable
-		};
-		this.props.editMapDetails(editedMap);
-	}
-
-	private formatDisplayable() {
-		let styleFn;
-		let messageId;
-		let buttonMessageId;
-
-		if (this.props.map.displayable) {
-			styleFn = this.styleEnabled;
-			messageId = 'map.is.displayable';
-			buttonMessageId = 'hide';
-		} else {
-			styleFn = this.styleDisabled;
-			messageId = 'map.is.not.displayable';
-			buttonMessageId = 'show';
-		}
-
-		let toggleButton;
-		if (hasToken()) {
-			// throw out alert if the admin wants to display uncalibrated map
-			if (!(this.props.map.origin && this.props.map.opposite)) {
-				toggleButton = <Button style={this.styleToggleBtn()} color='primary' onClick={this.notifyCalibrationNeeded}>
-					<FormattedMessage id={buttonMessageId} />
-				</Button>;
-			}
-			// if map is already calibrated, the button will allow it to be displayed
-			else {
-				toggleButton = <Button style={this.styleToggleBtn()} color='primary' onClick={this.toggleMapDisplayable}>
-					<FormattedMessage id={buttonMessageId} />
-				</Button>;
-			}
-		} else {
-			toggleButton = <div />;
-		}
-
-		return (
-			<span>
-				<span style={styleFn()}>
-					<FormattedMessage id={messageId} />
+const MapViewComponent: React.FC<MapViewProps> = ({ mapID }) => {
+	const [showEditModal, setShowEditModal] = useState(false);
+	const handleShowModal = () => setShowEditModal(true);
+	const handleCloseModal = () => setShowEditModal(false);
+
+	const selectMapById = React.useMemo(makeSelectMapById, []);
```

Comment:
What's the rationale for using a factory method here for the selector? Reselect generally doesn't require this anymore for multiple instances ever since their introduced weakMap memoization.

 See  https://reselect.js.org/introduction/v5-summary#breaking-changes

Reply:
This is selecting the map by ID. I used a [selector factory](https://redux.js.org/usage/deriving-data-selectors#selector-factories) as it seemed to be the best option at the time. If there is a better way to do this please let me know.

Reply:
Looking more closely at the selectors, this should not require a memoized createSelector . All it's doing is extracting the selected id from the map state. Since there is no data manipulation, or derivation it will always return a 'stable reference' to the map state entity. So a simple selector will suffice.

```
export const selectMapById= (state: RootState,id: number) => state.maps.byMapId[id];
```

P.S. Side note, even if this did need memoization, there was a newer update that essentially gets rid of the need for the factorySelector functions + useMemo() combo.
[WeakMapMemoize](https://reselect.js.org/api/weakMapMemoize) is now the baseline default 'memoize' function. Meaning that one instance used across multiple components will be memoized properly!

Reply:
Thank you for the explanation, changes have been made on commit [811dc66](https://github.com/OpenEnergyDashboard/OED/pull/1314/commits/811dc6693629b4bef6f78de78981095a93f66e0d)